src/components/AppBar/AppBar.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+	import type { Snippet } from "svelte";
+
+	interface AppBarProps {
+		/* Styling */
+		rootBackground?: string;
+		rootBorder?: string;
+		rootPadding?: string;
+		rootShadow?: string;
+		rootSpacing?: string;
+		rootGridColumns?: string;
+		rootGap?: string;
+		regionRowMain?: string;
+		regionRowHeadline?: string;
+		leadSnippetClasses?: string;
+		defaultSnippetClasses?: string;
+		trailSnippetClasses?: string;
+
+		/* Snippets */
+		regionRowHeadlineSnippet?: Snippet;
+		leadSnippet?: Snippet;
+		children?: Snippet;
+		trailSnippet?: Snippet;
+
+		/* Accessibility */
+		label?: string;
+		labelledby?: string;
+
+		/* Rest */
+		class?: string;
+	}
+
+	let {
+		rootBackground = "bg-surface-100-800-token",
+		rootBorder = '',
+		rootPadding = 'p-4',
+		rootShadow = '',
+		rootSpacing = "space-y-4",
+		rootGridColumns = "grid-cols-[auto_1fr_auto]",
+		rootGap = "gap-4",
+		regionRowMain = '',
+		regionRowHeadline= '',
+		leadSnippetClasses= '',
+		defaultSnippetClasses= '',
+		trailSnippetClasses= '',
+		regionRowHeadlineSnippet,
+		leadSnippet,
+		children: defaultSnippet,
+		trailSnippet,
+		label = '',
+		labelledby = '',
+		/* Unsure what to alias this as "class" is reserved in js, alternatives: className, restClass, restClassName, clazz */
+		class: className = ''
+	} = $props<AppBarProps>();
+
+	const joined = (...items: string[]) => items.join(" ");
+	const cBase = "flex flex-col";
+	const cRowMain = "grid items-center";
+	const cLeadSnippet = "flex-none flex justify-between items-center";
+	const cDefaultSnippet = "flex-auto";
+	const cTrailSnippet = "flex-none flex items-center space-x-4";
+
+	const classesBase = $derived(joined(cBase, rootBackground, rootBorder, rootSpacing, rootPadding, rootShadow, className));
```

Comment:
We're also doing away with the reactive grouping `$: ...` or `$derived()` in this case in favor of just dropping the style props directly into the template:
https://github.com/skeletonlabs/skeleton-next/blob/main/src/components/Card/Card.svelte#L65

If we need to have some kind of dynamic swapping, like a ternary that chooses between prop A and B. We handle it like so:

```ts
{
   fooActive: '...',
   fooInactive: '...'
}

const _fooActive = someActiveState ? fooActive : fooInactive;
```
> NOTE: I'm not sold on the `_` convention, but it's a bare minimum example.

Then in the template we would use this new class:
```html
<div class="{fooBase} ... {_fooActive} {fooRest}" />
```

So we intercept and modify ONLY when needed. But by default we try to pass the style props directly to the template when we can!